R/sampler.R

```diff
@@ -49,28 +51,25 @@ particle_filter <- function(data, model, compare, n_particles,
   }
 
   i_state <- seq_along(model$initial()) + 1L
-
-  ## Special treatment for the burn-in phase; later we might use this
-  ## same approach for skipping steps though.
-  if (save_particles) {
+  
+  if (save_particles){
     ## Storage for all particles:
     particles <- array(NA_real_,
                        c(max(data$day_end) + 1L + forecast_days,
                          length(i_state), n_particles))
-    step <- seq(data$step_start[[1L]], data$step_end[[1L]],
-                attr(data, "steps_per_day"))
-    state_with_history <-
-      model$run(step, use_names = FALSE, replicate = n_particles)
-    particles[seq_len(data$day_end[[1]] + 1), , ] <-
-      state_with_history[, i_state, ]
-    state <- state_with_history[length(step), i_state, , drop = TRUE]
   } else {
     particles <- NULL
-    step <- c(data$step_start[[1L]], data$step_end[[1L]])
-    state <- model$run(step = step, use_names = FALSE, replicate = n_particles,
-                       return_minimal = TRUE)[, 1, , drop = TRUE]
   }
 
+  X <- pre_data_run(model = model,
+                    data = data,
+                    seeding_func = seeding_func, 
+                    n_particles = n_particles,
+                    i_state = i_state,
+                    particles = particles)
```

Comment:
Does this need to check that the first date in data is after the time being run up to, or will it work in either case?

Reply:
I think this depends on whether the seeding_func is set up correctly. I've done a newer version of seeding.R which uses a truncated poisson function (thus avoiding a potentially long while loop) and makes clearer that the seeding times must be before the first data time point